test/middleware/PostQuery.test.ts

```diff
@@ -0,0 +1,53 @@
+import 'jest';
+import PostQuery from '../../src/middleware/PostQuery';
+
+describe('Test API key header verification', () => {
+    let mockURI: string;
+    let mockProjectID: string;
+    let mockQueryData: QueryData;
+    let postQuery: PostQuery;
+
+    beforeEach(() => {
+        /* The mock data below is pulled from a personal development dB.
+
+       To perform some of these tests yourself, you must connect your dB
+       to the webapp project and create a mock project with the webapp,
+       then change these values to match the new data.
+       */
+        mockURI = 'http://localhost:3000';
```

Comment:
Let's create an issue to look into mock possibilities to avoid the use of a database